src/ucp/tag/eager_snd.c

```diff
@@ -258,6 +180,9 @@ static ucs_status_t ucp_tag_eager_bcopy_single(uct_pending_req_t *self)
                                                  ucp_tag_pack_eager_dt);
     if (status == UCS_OK) {
         ucp_request_t *req = ucs_container_of(self, ucp_request_t, send.uct);
+        if (UCP_DATATYPE_GENERIC == (req->send.datatype & UCP_DATATYPE_CLASS_MASK)) {
```

Comment:
shall we add inline function ucp_request_dt_finish() to contain these 3 lines (which also repeat in ucp_tag_eager_bcopy_multi and ucp_tag_eager_sync_bcopy_single)